configs/myconfigs/sample_one_var.py

```diff
@@ -0,0 +1,223 @@
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Authors: Jason Lowe-Power, Brian Coutinho
+
+""" This script shows how one could dynamically run more simulations
+    at single sample point. This an improved version of sample_one.py
+    We measure the variance in a certain stat after running an intial
+    number of simulations. If it is not below a fixed bound we run
+    additional simulations.
+
+    To use this, first generate sample points with runkvm.py.
+    For ex. setting the output directory to ./examples/bench1/ and
+    running runkvm.py should generate a file
+    ./examples/bench1/random.samples
+
+    To simulate a specific sample number simply run this script
+    by pointing to the same output directory. This script will
+    automatically create a sub-direcotry for the sample.
+"""
+
+import os
+import sys
+import time
+
+import m5
+from m5.objects import *
+from m5.ticks import fromSeconds
+from m5.util.convert import toLatency
+
+sys.path.append('configs/common/') # For the next line...
+import SimpleOpts
+
+from system import MySystem
+
+# Sampling library
+from sampling import forwardToSample, Sample, makeSampleDir, \
+                     checkSystem, loadSamples
+# Sampling statistics
+from sampling import stats
+
+SimpleOpts.add_option("--script", default='',
+                      help="Script to execute in the simulated system")
+SimpleOpts.set_usage("usage: %prog [options] sampleno")
+
+
+# Global Constants
+
+# initruns is the minimum number of simulations we run at a sample
+#  use a suitable value that can discriminate between
+#  low and high variance points
+initruns = 10
+# maxruns bounds the number of simulations
+maxruns = 50
+# threshold on std error mean
+threshold = 0.01
+
+def simulateROI(system, opts, sampleno):
+    """ Fast forward to and simulate a single sample point
+        After running a initial number of simulations check
+        if the variance is below a set threshold. If not
+        run more simulations until a maximum limit
+
+        Uses global parameters defined above
+    """
+
+    # Create the sample directory
+    sample_dir = makeSampleDir(sampleno)
+
+    # First forward to the specific sample number
+    print "Fast forwarding to sample", sampleno
+    forwardToSample(system, sampleno)
+
+    # Take an intial number of simulations/runs
+    Sample(system, opts, initruns)
+
+    # Check variance on a chosen stat, in this case
+    # we are using floating point instructions
```

Comment:
Also add:
# You may want to change this to another statistic of interest for your application